### PR TITLE
fix lock directory being created with incorrect permissions

### PR DESCRIPTION
**Pull request by nnutter**
Partial fix for [RT#105060](https://rt.gsc.wustl.edu/Ticket/Display.html?id=105060).

### CODE REVIEW

File: lib/perl/Genome/Sys/Lock/FileBackend.pm [hunk 1]
```diff
@@ -276,17 +277,11 @@ sub lock_dir_for_resource {
...
-    my $obj = Path::Class::dir($dir_path);
-    $obj->mkpath(0, 0777);
-}
-
 sub tempdir_for_resource {
     my ($self, $resource_lock) = @_;
 
     my $lock_dir = $self->lock_dir_for_resource($resource_lock);
-    $self->make_dir_path($lock_dir);
+    Genome::Sys->create_directory($lock_dir . '');
```
Comment (acoffman) [inline lib/perl/Genome/Sys/Lock/FileBackend.pm]: My best guess is that this is here to force `$lock_dir` to stringify, but it [appears](https://github.com/nnutter/genome/blob/lock-pair/lib/perl/Genome/Sys/Lock/FileBackend.pm#L277) that we already have a string. Am I missing something? (Highly likely!). If it is to force stringfication, is there a reason to prefer this to `"$lock_dir"`?

File: lib/perl/Genome/Sys/Lock/FileBackend.pm [hunk 2]
```diff
@@ -300,8 +295,7 @@ sub tempdir_for_resource {
         Carp::croak("Failed to create temp lock directory ($tempdir)");
     }
 
-    # make this readable for everyone
-    chmod(0770, $tempdir) or Carp::croak("Can't chmod 0770 path ($tempdir): $!");
+    mode($tempdir)->add_group_rwx();
```
Comment (acoffman) [inline lib/perl/Genome/Sys/Lock/FileBackend.pm]: It probably doesn't actually matter, but do we want group execute permissions or just r/w?

Comment (nnutter) [inline lib/perl/Genome/Sys/Lock/FileBackend.pm]: It's a directory so you want execute.

Comment (nnutter) [inline lib/perl/Genome/Sys/Lock/FileBackend.pm]: You arguably don't need write.

File: lib/perl/Genome/Sys/Lock/FileBackend.pm (hunk 1 above)
Comment (nnutter) [inline lib/perl/Genome/Sys/Lock/FileBackend.pm]: You're right, I thought it was a `Path::Class::File`.  If it were you can `"$lock_dir"`, `$lock_dir . ''`, or `$lock_dir->stringify`.  TIMTOWTDI.

Comment (nnutter) [inline lib/perl/Genome/Sys/Lock/FileBackend.pm]: Oh, and the forced stringification... it's paranoia and I probably shouldn't have bothered.

File: lib/perl/Genome/Sys/Lock/FileBackend.pm (hunk 2 above)
Comment (acoffman) [inline lib/perl/Genome/Sys/Lock/FileBackend.pm]: Ah yes, you're totally correct!

**Comment by acoffman** [comment]
+1 :lock: :file_folder:

**Comment by tmooney** [comment]
+1 :swimmer:

**Pull request merged**